components/infra-proxy-service/server/environments.go

```diff
@@ -143,7 +142,65 @@ func (s *Server) DeleteEnvironment(ctx context.Context, req *request.Environment
 	return &response.Environment{
 		Name: environment.Name,
 	}, nil
+}
+
+// UpdateEnvironment updates the environment attributes
+func (s *Server) UpdateEnvironment(ctx context.Context, req *request.UpdateEnvironment) (*response.Environment, error) {
+	c, err := s.createClient(ctx, req.OrgId, req.ServerId)
+	if err != nil {
+		return nil, err
+	}
+
+	if req.Name == "" {
+		return nil, status.Error(codes.InvalidArgument, "must supply environment name")
+	}
+
+	cookbooks := req.CookbookVersions
+	if len(cookbooks) == 0 {
+		cookbooks = map[string]string{}
+	}
```

Comment:
This reads a little funny, since `len(map[string]string{})` is `0`, too: [playground](https://play.golang.org/p/uzCsFzkmkkz).

I'd propose this
```suggestion
	if cookbooks == nil {
		cookbooks = map[string]string{}
	}
```

Reply:
Ouch, I was left with len check. Earlier it was Array, I have changed it to map string. Well, thanks!